src/connectors/hits/connectHits.js

```diff
@@ -99,8 +100,22 @@ export default function connectHits(renderFn, unmountFn) {
         );
       },
 
-      dispose() {
+      dispose({ state }) {
         unmountFn();
+
+        if (!escapeHTML) {
+          return state;
+        }
+
+        return state.setQueryParameters(
+          Object.keys(TAG_PLACEHOLDER).reduce(
```

Comment:
Object.fromEntries would make this cleaner, but needs to be polyfilled, so this lgtm